strongbox-data-service/src/main/java/org/carlspring/strongbox/gremlin/tx/GraphTransaction.java

```diff
@@ -0,0 +1,123 @@
+package org.carlspring.strongbox.gremlin.tx;
+
+import java.util.Iterator;
+import java.util.Optional;
+
+import org.apache.commons.configuration.Configuration;
+import org.apache.tinkerpop.gremlin.process.computer.GraphComputer;
+import org.apache.tinkerpop.gremlin.process.traversal.TraversalSource;
+import org.apache.tinkerpop.gremlin.process.traversal.dsl.graph.GraphTraversalSource;
+import org.apache.tinkerpop.gremlin.structure.Edge;
+import org.apache.tinkerpop.gremlin.structure.Graph;
+import org.apache.tinkerpop.gremlin.structure.Transaction;
+import org.apache.tinkerpop.gremlin.structure.Vertex;
+import org.apache.tinkerpop.gremlin.structure.io.Io;
+import org.apache.tinkerpop.gremlin.structure.io.Io.Builder;
+import org.neo4j.ogm.session.Session;
+import org.neo4j.ogm.session.SessionFactory;
+import org.opencypher.gremlin.neo4j.ogm.transaction.GremlinTransaction;
+import org.springframework.data.neo4j.transaction.SessionHolder;
+import org.springframework.transaction.support.TransactionSynchronizationManager;
+
+/**
+ * Exposes current thread bound {@link Graph} transaction.
+ *
+ * @author sbespalov
+ */
+public class GraphTransaction implements Graph
+{
+    private static final Class<SessionHolder> sessionHolderClass = SessionHolder.class;
+    private static final Class<GremlinTransaction> gremlinTransactionClass = GremlinTransaction.class;
+
+    private final SessionFactory sessionFactory;
+
+    public GraphTransaction(SessionFactory sessionFactory)
+    {
+        this.sessionFactory = sessionFactory;
+    }
+
+    public Vertex addVertex(Object... keyValues)
+    {
+        return getCurrent().addVertex(keyValues);
+    }
+
+    public Vertex addVertex(String label)
+    {
+        return getCurrent().addVertex(label);
+    }
+
+    public <C extends GraphComputer> C compute(Class<C> graphComputerClass)
+        throws IllegalArgumentException
+    {
+        return getCurrent().compute(graphComputerClass);
+    }
+
+    public GraphComputer compute()
+        throws IllegalArgumentException
+    {
+        return getCurrent().compute();
+    }
+
+    public <C extends TraversalSource> C traversal(Class<C> traversalSourceClass)
+    {
+        return getCurrent().traversal(traversalSourceClass);
+    }
+
+    public GraphTraversalSource traversal()
+    {
+        return getCurrent().traversal();
+    }
+
+    public Iterator<Vertex> vertices(Object... vertexIds)
+    {
+        return getCurrent().vertices(vertexIds);
+    }
+
+    public Iterator<Edge> edges(Object... edgeIds)
+    {
+        return getCurrent().edges(edgeIds);
+    }
+
+    public Transaction tx()
+    {
+        return getCurrent().tx();
+    }
+
+    public void close()
+        throws Exception
+    {
+        getCurrent().close();
+    }
+
+    public <I extends Io> I io(Builder<I> builder)
+    {
+        return getCurrent().io(builder);
+    }
```

Comment:
Just for my better understanding, what is this `io` used for exactly...?

Reply:
this is just delegates to `org.apache.tinkerpop.gremlin.structure.Graph#io`, actually all methods in this class just delegate to current (transaction `ThreadLocal` bound) `Graph` instance.

Reply:
and there is already JavaDocs for this class :)